cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -252,3 +259,34 @@ func (nm *nodeMap) gsaEmailsByNode(node string) ([]gsaEmail, error) {
 	}
 	return l, nil
 }
+
+type nodeZones struct {
+	sync.RWMutex
+	m      map[string]string
+	client clientset.Interface
+}
+
+func newNodeZones(client clientset.Interface) *nodeZones {
+	return &nodeZones{
+		m:      make(map[string]string),
+		client: client,
+	}
+}
+
+func (nz *nodeZones) zoneByNode(name string) (string, error) {
+	nz.Lock()
+	defer nz.Unlock()
+	if zone, ok := nz.m[name]; ok {
```

Comment:
should this be nz.RLock() since we are checking if the value exists?

Reply:
Changed to sync.Mutex.